Sources/Controllers/OARootViewController.m

```diff
@@ -696,4 +699,43 @@ - (void) popoverControllerDidDismissPopover:(UIPopoverController *)popoverContro
     }
 }
 
+- (BOOL)canBecomeFirstResponder {
+    return YES;
+}
+
+- (NSArray *)keyCommands {
+    return @[[UIKeyCommand keyCommandWithInput:UIKeyInputDownArrow modifierFlags:0 action:@selector(zoomOut) discoverabilityTitle:OALocalizedString(@"key_hint_zoom_out")],
+             [UIKeyCommand keyCommandWithInput:UIKeyInputUpArrow modifierFlags:0 action:@selector(zoomIn) discoverabilityTitle:OALocalizedString(@"key_hint_zoom_in")],
+             [UIKeyCommand keyCommandWithInput:UIKeyInputEscape modifierFlags:0 action:@selector(goBack) discoverabilityTitle:OALocalizedString(@"key_hint_goback")],
+             [UIKeyCommand keyCommandWithInput:@"\r" modifierFlags:0 action:@selector(chooseSelection) discoverabilityTitle:OALocalizedString(@"key_hint_select")]];
+}
+
+- (void)zoomOut {
+    if ([OAAppSettings sharedManager].settingExternalInputDevice != NO_EXTERNAL_DEVICE) {
+        [_mapViewController animatedZoomOut];
+        [_mapViewController calculateMapRuler];
+    }
+}
+
+- (void)zoomIn {
+    if ([OAAppSettings sharedManager].settingExternalInputDevice != NO_EXTERNAL_DEVICE) {
+        [_mapViewController animatedZoomIn];
+    }
+}
+
+- (void)goBack {
+    if ([OAAppSettings sharedManager].settingExternalInputDevice == WUNDERLINQ_EXTERNAL_DEVICE) {
+        //Launch WunderLINQ
+        NSString *wunderlinqAppURL = @"wunderlinq://datagrid";
+        BOOL canOpenURL = [[UIApplication sharedApplication]
+                       canOpenURL:[NSURL URLWithString:wunderlinqAppURL]];
+        if ( canOpenURL ) [[UIApplication sharedApplication]
+                       openURL:[NSURL URLWithString:wunderlinqAppURL] options:@{} completionHandler:nil];
+    }
+}
+
+- (void)chooseSelection {
+    printf("Enter\n");
```

Comment:
Is it needed?

Reply:
Sorry, that was for future use.  I'll remove.